src/js/gi/containers/Modals.jsx

```diff
@@ -0,0 +1,288 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import * as actions from '../actions';
+import Modal from '../../common/components/Modal';
+
+export class Modals extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.renderProfilePageModals = this.renderProfilePageModals.bind(this);
+    this.renderProfileSchoolHeaderModals = this.renderProfileSchoolHeaderModals.bind(this);
+    this.renderProfileVeteranSummaryModals = this.renderProfileVeteranSummaryModals.bind(this);
+    this.renderProfileSummaryModals = this.renderProfileSummaryModals.bind(this);
+    this.renderProfileHistoryModals = this.renderProfileHistoryModals.bind(this);
+    this.renderProfileCautionFlagModals = this.renderProfileCautionFlagModals.bind(this);
+    this.renderProfileCalculatorModals = this.renderProfileCalculatorModals.bind(this);
+  }
+
+  renderProfilePageModals() {
+    return (
+      <span>
+        <Modal onClose={() => this.props.hideModal()} visible={!!this.props.giModals.retention}>
```

Comment:
for these onClose function with no parameters, I believe you can do `onClose={this.props.hideModal}`; no need to wrap it in a function